dark/fasta.py

```diff
@@ -94,15 +94,17 @@ def iter(self):
         instance of the desired read class.
         """
         for seq in SeqIO.parse(self.file_, 'fasta'):
-            yield self.readClass(seq.description, str(seq.seq))
+            read = self.readClass(seq.description, str(seq.seq))
+            if read.checkAlphabeth():
```

Comment:
I'm not sure I like it checking every sequence. It seems like just checking the first several would be fine. It seems to me like either the FASTA file is full of AA sequences or full of DNA sequences, rather than some kind of mix in which we'd have to worry about finding a non-AA sequence at the end of a file full of DNA.

Anyway, leave it like this for now & we can speed it up later if we need to. It's good to be safe (and simple), too.